lib/knapsack_pro/runners/queue/base_runner.rb

```diff
@@ -59,9 +59,37 @@ def trap_signals
             Signal.trap(signal) {
               puts "#{signal} signal has been received. Terminating Knapsack Pro..."
               @@terminate_process = true
+              log_threads
             }
           end
         end
+
+        def log_threads
+          threads = Thread.list
+
+          puts
+          puts '=' * 80
+          puts "Start logging #{threads.count} detected threads."
+          puts 'Use the following backtrace to find the line of code that got stuck if the CI node hung and terminated your tests.'
+
+          threads.each do |thr|
+            puts
+            if thr == Thread.main
+              puts "Main thread and its backtrace:"
```

Comment:
recommendation

Maybe we can use the same string as the one for the non-main thread?

```suggestion
              puts "Main thread backtrace:"
```